flocker/common/runner.py

```diff
@@ -256,7 +340,59 @@ def scp_failed(reason):
         """
         reason.trap(ProcessTerminated)
         if failed_reason:
-            return Failure(failed_reason[0])
+            return Failure(failed_reason[-1])
+        return reason
+
+    with action.context():
+        context = DeferredContext(
+            run(
+                reactor,
+                scp_command,
+                handle_stdout=handle_stdout,
+                handle_stderr=handle_stderr,
+            )
+        )
+
+        context.addErrback(scp_failed)
 
-    scp_result.addErrback(scp_failed)
-    return scp_result
+        return context.addActionFinish()
+
+
+def download(reactor, username, host, remote_path, local_path,
+             port=22, identity_file=None):
+    """
+    Run the local ``scp`` command to download a file or directory from a remote
+    host and kill it if the reactor stops.
+
+    See ``scp`` for parameter and return type documentation.
```

Comment:
:question: Could you have used kwargs here? seems like it might be less maintenance burden.